src/store/wizardSlice.ts

```diff
@@ -87,6 +94,11 @@ export type wizardState = {
     useLatest: boolean;
     snapshotDate: string;
   };
+
+  userState: {
+    users: UserWithAdditionalInfo[];
+    userInfo: UserWithAdditionalInfo;
+  };
```

Comment:
What blocks us from doing just:
```suggestion
  users: UserWithAdditionalInfo[];
```
and accessing the users by their name value?

This way we have a complex structure that includes an array and then just one item of the array that is stored outside of it

Reply:
it becomes very complicated to implement the change of every field. and like in this way, the verifaition of every filed stay simple